### PR TITLE
Bump NumPy to 1.21

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

I assume that 3.8 will not be released before June 23...

https://numpy.org/neps/nep-0029-deprecation_policy.html

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by ksunden — APPROVED** [review]

**Review by timhoffm — CHANGES_REQUESTED** [review]
Is the numpy support date the relevant number?

We state (https://matplotlib.org/stable/devel/min_dep_policy.html):

> All minor versions of numpy released in the 24 months prior to the project, and at minimum the last three minor versions.

1.21.6 was released on 12. Apr. 2022 (https://pypi.org/project/numpy/#history) so we cannot bump to 1.22 before Apr. 2024.

But I think we can bump to 1.21. The last 1.20.x release was in May 2021.

**Comment by oscargus** [comment]
I was under the impression that we followed NEP29 and that interpretation: 

"On Jun 23, 2023 drop support for NumPy 1.21 (initially released on Jun 22, 2021)"

However, the formulation is slightly different there:

"all minor versions of NumPy released in the prior 24 months from the anticipated release date with a minimum of 3 minor versions of NumPy."

I'll put into draft until the dev call on Thursday.

**Comment by timhoffm** [comment]
I'm not clear either how the statement is to be understood. Please clarify in the call on Thursday. Anybody can dismiss my comment afterwards.

**Comment by oscargus** [comment]
From devcall: it is the first version, so strictly we can go with 1.22. However, as the next release is expected around June 23, we shouldn't really push it, so makes sense to go with 1.21.

File: doc/api/next_api_changes/development/24919-KS.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@ being bumped:
 +------------+-----------------+---------------+
 | kiwisolver |       1.0.1     |      1.3.1    |
 +------------+-----------------+---------------+
+|   numpy    |       1.20.0    |      1.21.0   |
```
Comment (QuLogic) [inline doc/api/next_api_changes/development/24919-KS.rst:14]:
```suggestion
|   NumPy    |       1.20.0    |      1.21.0   |
```

**Review by timhoffm — APPROVED** [review]

**Pull request merged**